data/frameProcessor/include/MercuryAdditionPlugin.h

```diff
@@ -0,0 +1,58 @@
+/*
+ * MercuryAdditionPlugin.h
+ *
+ *  Created on: 15 Oct 2020
+ *      Author: Christian Angelsen
+ */
+
+#ifndef INCLUDE_MERCURYADDITIONPLUGIN_H_
+#define INCLUDE_MERCURYADDITIONPLUGIN_H_
+
+#include "MercuryProcessorPlugin.h"
+
+namespace FrameProcessor
+{
+  /** Applies the Charged Sharing algorithm to a Mercury frame.
+   *
+   * The MercuryAdditionPlugin examines surrounding neighbouring pixels
+   * moving any event shared across multiple pixels onto the pixel containing
+   * the biggest portion of that event.
+   */
+  class MercuryAdditionPlugin : public MercuryProcessorPlugin
+  {
+    public:
+      MercuryAdditionPlugin();
+      virtual ~MercuryAdditionPlugin();
+
+      void configure(OdinData::IpcMessage& config, OdinData::IpcMessage& reply);
+      void requestConfiguration(OdinData::IpcMessage& reply);
+      void status(OdinData::IpcMessage& status);
+      bool reset_statistics(void);

```

Comment:
My knowledge of C++'s inheritance is a bit hazy, but I don't think I can do that. If I (let's say) remove the configure(..) declaration from a plugin's header, I get an error saying no member function declared in class for configure(). Because the configure() function won't be the same across each plugin, I have to declare and define it in each.

Although only MercuryReorderPlugin's reset_statistics() differ from the rest of the plugins'. So I could remove the declaration and definition for reset_statistics() for eight out of the nine plugins. Would it be worthwhile doing that, or better to keep reset_statistics() defined and declared across all plugins for consistency?

For reference, the other three functions all differ from one plugin to the next so I think they'll have to be declared as they are.

Reply:
Ah understandable, I don't fully understand how the inheritance works either, so I wasn't sure.

You certainly cannot inherit the actual definition from the cpp files, but the declarations in the headers might be inheritable